src/int_32.ts

```diff
@@ -32,6 +34,38 @@ export class Int32 extends BSONValue {
     this.value = +value | 0;
   }
 
+  /**
+   * Attempt to create an Int32 type from string.
+   *
+   * This method will throw a BSONError on any string input that is not representable as an Int32.
+   * Notably, this method will also throw on the following string formats:
+   * - Strings in non-decimal formats (exponent notation, binary, hex, or octal digits)
+   * - Strings non-numeric and non-leading sign characters (ex: '2.0', '24,000')
+   * - Strings with leading and/or trailing whitespace
+   *
+   * Strings with leading zeros, however, are also allowed
+   *
+   * @param value - the string we want to represent as an int32.
+   */
+  static fromString(value: string): Int32 {
+    const cleanedValue = !/[^0]+/.test(value)
+      ? value.replace(/^0+/, '0') // all zeros case
+      : value[0] === '-'
+        ? value.replace(/^-0+/, '-') // negative number with leading zeros
+        : value.replace(/^\+?0+/, ''); // positive number with leading zeros
+
+    const coercedValue = Number(value);
+    if (
+      coercedValue.toString() !== cleanedValue ||
+      !Number.isSafeInteger(coercedValue) ||
+      BSON_INT32_MAX < coercedValue ||
+      BSON_INT32_MIN > coercedValue
+    ) {
+      throw new BSONError(`Input: '${value}' is not a valid Int32 string`);
+    }
```

Comment:
Could we throw specific errors here?  Seems like low hanging fruit to give users a better experience.  For example, we could throw something like the following if the value is > the int32 max: `Input ${value} is larger than the maximum value for Int32`

Reply:
Sounds good, I've update the error messages, but for the catch all `toString() ===` check, I've kept the generic message.